jax/_src/scipy/eigh.py

```diff
@@ -30,6 +56,62 @@ def _similarity_transform(
   return jnp.dot(out, matrix_out, precision=precision)
 
 
+def _canonically_purify_initial_guess(H, rank):
+  lambda_min, lambda_max = _gershgorin(H)
+  N = H.shape[0]
+  mu = jnp.trace(H) / N
+  theta = rank / N
+  beta = theta / (lambda_max - mu)
```

Comment:
We should probably guard against division by zero here. This would blow up for H=eye(n), I think.